### PR TITLE
Support version 1.0 of simple-salesforce for login

### PR DESCRIPTION
**Pull request by lambacck**
The `sandbox` parameter is no longer supported by simple-salesforce. This change keeps the `sandbox` parameter for salesforce-bulk while also introducing the `domain` parameter from simple-salesforce. If the `sandbox` parameter is provided it will be converted to the a `domain` parameter value of `'test'`, any value passed to `domain` will be passed straight through to simple-salesforce as is.

Closes: #85
Closes: #80 
Closes: #63 
Closes: #81 
Closes: #76

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -20,9 +20,9 @@
 
 requires = [
     'six',
-    'requests>=2.2.1',
+    'requests>=2.24.0',
```
Comment (sigmavirus24) [inline setup.py:23]: We have this and `requirements.txt` and use tox. Tox should respect these dependencies when testing and this is all pip cares about. Why do we maintain both?

Further I see a `PublishCommand` but we should be using `twine` + PyPI Upload Tokens.  Should we switch to that here too?

Comment (lambacck) [reply setup.py:23]: because travisci doesn't uses requirements.txt and not tox :(

Comment (lambacck) [reply setup.py:23]: Also PublishCommand is using twine.

Comment (sigmavirus24) [reply setup.py:23]: > because travisci doesn't uses requirements.txt and not tox :(

I don't understand this

**Comment by sumanth232** [comment]
HI,
Can someone release a new tagged version with this fix ?
The current version v2.1.0 is unusable due to its unpinned dependancy (simple-salesforce>0.69). 
'pip install salesforce-bulk' downloads the latest version of simple-salesforce which is not compatible with v2.1.0

**Pull request merged**